src/ESLintLinter.php

```diff
@@ -166,11 +166,16 @@ protected function parseLinterOutput($path, $err, $stdout, $stderr) {
         if (strpos($offense['message'], "File ignored ") === 0) {
           continue;
         }
-
+        
+        $ruleid = $offense['ruleId'];
+        if(!$ruleid){
+            $ruleid = 'unknown';
+        }
+        
         $message = new ArcanistLintMessage();
         $message->setPath($file['filePath']);
         $message->setSeverity($this->mapSeverity($offense['severity']));
-        $message->setName($offense['ruleId'] || 'unknown');
+        $message->setName($ruleid);
```

Comment:
libphutil (packaged with arcanist) has a nicer pattern for this:

```suggestion
        $message->setName(idx($offense, 'ruleid', 'unknown'));
```